frontend/src/services/applicationService.js

```diff
@@ -257,14 +267,11 @@ export default {
     return fundingAgreement?.stateCode === CRM_STATE_CODES.ACTIVE && isUnexpired
   },
 
-  async hasActiveApplicationOrFundingAgreement(facilities = []) {
+  async hasApprovedApplication(facilities = []) {
     const results = await Promise.all(
       facilities.map(async (facility) => {
-        const applications = await this.getActiveApplicationsByFacilityId(facility.facilityId)
-        return applications?.some(async (application) => {
-          application.fundingAgreement = await FundingAgreementService.getActiveFundingAgreementByApplicationId(application.applicationId)
-          return this.checkApplicationStatus(application) || this.checkFundingAgreement(application?.fundingAgreement)
-        })
+        const applications = await this.getApprovedApplicationsByFacilityId(facility.facilityId)
```

Comment:
We don't actually need the applications here...just a count so we could improve performance by not returning them all from the backend (since we're stuck getting them all from CRM). We did something similar with reportService.getDraftSurveyResponsesCountByFacility. You could also pass in a list directly and reduce the number of API calls.

Reply:
good point! I will update it

Reply:
it's updated